### PR TITLE
added BaseFunction class with itemgetter functionality to retrieve parameters provided with a ParProxy sub-class similar to BinProxy

### PR DESCRIPTION
**Pull request by mverzett**

### CODE REVIEW

**Comment by ndawe** [comment]
Same thing here regarding travis. Will migrate the ROOT builds back to dropbox (copy.com was shut down)

File: rootpy/plotting/func.py [hunk 1]
```diff
@@ -14,9 +14,65 @@
...
+        @property
+        def index(self):
+            return self.idx_
+
+        @property
+        def name(self):
+            return self.fcn_.GetParName(self.idx_)
+        
+        @name.setter
+        def set_name(self, val):
```
Comment (ndawe) [inline rootpy/plotting/func.py]: How about just name instead of set_name?

File: rootpy/plotting/func.py [hunk 2]
```diff
@@ -14,9 +14,65 @@
...
+            return self.fcn_.SetParName(self.idx_, val)
+
+        @property
+        def value(self):
+            return self.fcn_.GetParameter(self.idx_)
+
+        @value.setter
+        def set_value(self, val):
```
Comment (ndawe) [inline rootpy/plotting/func.py]: Same here. set_value -> value

File: rootpy/plotting/func.py [hunk 3]
```diff
@@ -14,9 +14,65 @@
...
+            self.fcn_.SetParameter(self.idx_, val)
+
+        @property
+        def error(self):
+            return self.fcn_.GetParError(self.idx_)
+
+        @error.setter
+        def set_error(self, val):
```
Comment (ndawe) [inline rootpy/plotting/func.py]: set_error -> error

**Comment by ndawe** [comment]
+1 for this, pending minor changes. If you rebase on upstream master then you should have green travis tests.

**Comment by mverzett** [comment]
Rebased and comments implemented. Question: is it worth adding test units somewhere? If so, where?

**Comment by ndawe** [comment]
Thanks @mverzett ! Finally merging this. Regarding unit tests, they should eventually go in a test_func.py in rootpy/plotting. As you can see, unit test coverage in rootpy isn't like it is in root_numpy. I'd like to build coverage in a separate set of PRs. It's been on the to-do list... For now I can merge this as it is.

**Pull request merged**